plugins/hls-hlint-plugin/hls-hlint-plugin.cabal

```diff
@@ -69,41 +56,11 @@ library
     , transformers
     , unordered-containers
     , apply-refact          >=0.9.0.0
-    -- can be removed if https://github.com/ndmitchell/hlint/pull/1325#issue-1077062712 is merged
-    -- and https://github.com/haskell/haskell-language-server/pull/2464#issue-1077133441 is updated
-    -- accordingly
+    , ghc-lib
+    , ghc-lib-parser
     , ghc-lib-parser-ex
 
-  if (flag(hlint34))
-    -- This mirrors the logic in hlint.cabal for hlint-3.3
-    -- https://github.com/ndmitchell/hlint/blob/d3576de4529d8df6cca5a345f5b7e04474ff7bff/hlint.cabal#L79-L88
-    -- so we can make sure that we do the same thing as hlint
-    build-depends: hlint      ^>=3.4
-
-    if (!flag(ghc-lib) && impl(ghc >=9.0.1) && impl(ghc <9.1.0))
-      build-depends: ghc      ==9.0.*
-    else
-      build-depends:
-        , ghc-lib            ^>=9.2
-        , ghc-lib-parser-ex  ^>=9.2
-        , ghc-lib-parser     ^>=9.2
-
-      cpp-options:   -DHLINT_ON_GHC_LIB
-
-  else
-    -- This mirrors the logic in hlint.cabal for hlint-3.2
-    -- https://github.com/ndmitchell/hlint/blob/c7354e473c7d09213c8adc3dc94bf50a6eb4a42d/hlint.cabal#L79-L88
-    build-depends: hlint        ^>=3.2
-    if (!flag(ghc-lib) && impl(ghc >=8.10.1) && impl(ghc < 8.11.0))
-      build-depends: ghc >=8.10 && <9.0
-    else
-      build-depends:
-        , ghc
-        , ghc-lib            ^>=8.10.7.20210828
-        , ghc-lib-parser-ex  ^>=8.10
-
-      cpp-options:   -DHLINT_ON_GHC_LIB
-
+  cpp-options:   -DHLINT_ON_GHC_LIB
```

Comment:
I do think we should remove this and its uses before we merge this.

Reply:
I wonder if this option is still relevant for some users of ghcide as a library. 

Sigma IDE doesn't currently integrate the hlint plugin, but if it did we'd almost certainly want to use the non ghc-lib version

Reply:
I'm confused. There is no option now, we just unconditionally set this. If you want to preserve the flag (and the corresponding CPP), then sure, but I don't see why we'd have this half-way version?

Reply:
Of course, good point. For now let's preserve the flag with a default of True